python/pylibcudf/pylibcudf/libcudf/io/avro.pxd

```diff
@@ -47,3 +48,8 @@ cdef extern from "cudf/io/avro.hpp" \
     cdef cudf_io_types.table_with_metadata read_avro(
         avro_reader_options &options
     ) except +libcudf_exception_handler
+
+    cdef cudf_io_types.table_with_metadata read_avro(
```

Comment:
Do we need to duplicate all these declarations with and without a stream parameter? Could we just add the stream parameter to the existing API binding, and force callers to provide it? What stability guarantees do we have for pylibcudf's libcudf bindings? Even if this is a breaking change at the Cython layer to add a parameter, it seems like it should be okay to do. It will not be breaking to the pylibcudf Python API.

(May need input from @vyasr here.)

Reply:
> Do we need to duplicate all these declarations with and without a stream parameter?

If I look at the C++ I see
```cpp
table_with_metadata read_parquet(
  parquet_reader_options const& options,
  rmm::cuda_stream_view stream      = cudf::get_default_stream(),
  rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref());
```
We might be able to provide one declaration in cython with. We just need to expose `get_default_stream` to cython.
```python
cdef table_with_metadata read_parquet(
    parquet_reader_options args,
    cuda_stream_view stream = get_default_stream(),
) except +libcudf_exception_handler
```

Reply:
I'd prefer to do this in a follow up if possible.

Reply:
The broader question that we need to address is https://github.com/rapidsai/cudf/issues/15198. The only way for Cython to expose C++ functions with default parameters is to treat the different versions as separate overloads. stream and mr are going to be by far the most annoying cases of this, but they are far from the only ones. I think that we should let Matt's PR proceed for now, with the understanding that we might break this early in 25.06. I'm not concerned about making stability promises on this part of the API right now; I would rather get it right while it is relatively unused and then freeze the API.

Let's discuss #15198 in more detail early in 25.06. I've added it to our board.